### ISSUE TITLE
Phrase Suggester breaks in 2.0

### ISSUE THREAD

**Issue by JnBrymn-EB** [elastic/elasticsearch#15675]
*overview*: a phrase suggester query that works in 1.5.1 breaks in 2.0.0-beta2 and 2.1.1 an it shouldn't be related to the new "everything's a phrase" approach to queries and filters.

The following uses the python elasticsearch client.
```python
body = {
    "mappings": {
      "movie": {
        "properties": {
          "genres": {
            "properties": {
              "name": { 
                "type": "string",
                "index": "not_analyzed"}}},
          "title": {
            "type": "string",
            "analyzer": "english",
            "copy_to":["suggestion"]},
          "suggestion": {
            "type": "string"}}}}}
es.indices.create("tmdb",body=body)
```
Then index some documents. Then:
```
query_body = { 
  "fields": ["title"],
  "query": {
    "match": {"title":"star trec"}},
  "suggest": { "title_completion": {
    "text": "star trec",
    "phrase": {
      "field": "suggestion",
      "max_errors": 2,
      "collate": {
        "query": { 
          "match_phrase": {
            "title" : "{{suggestion}}"
          }
        }}}}}}
        
es.search(index="tmdb",body=query_body,size=2)
```
The above works in 1.5.1 but not 2.0.0-beta2 or 2.1.1. The 2.1 documentation is roughly the same as 1.5 and indicates that this should work.

The error I get is
```
TransportError: TransportError(500, {u'failed_shards': [{u'node': u'nHzvQPtVR2KW7NMeOuvQLg', u'index': u'tmdb', u'reason': {u'reason': u'unexpected field [match_phrase]', u'type': u'script_parse_exception'}, u'shard': 0}], u'root_cause': [{u'reason': u'unexpected field [match_phrase]', u'type': u'script_parse_exception'}], u'grouped': True, u'reason': u'all shards failed', u'phase': u'query_fetch', u'type': u'search_phase_execution_exception'})
```

**Comment by s1monw** [comment]
can you elaborate what works and what doesn't? I wonder if you can try this with an index with only 1 shard just to make sure that the new hashing we use in 2.0 doesn't influence the results since this suggester really relies on per shard frequencies etc.?

**Comment by JnBrymn-EB** [comment]
Sorry @s1monw. I've added the error to the original comment to keep everything together. (I meant to include the error but apparently I got distracted and didn't add it.)

Also I retried the experiment with a single shard as you suggested and I had the same results.

**Comment by s1monw** [comment]
@JnBrymn-EB I see - this is a documentation problem. The syntax you are using is deprecated. I opened a PR to fix this: #15692  can you give it a try?

**Comment by JnBrymn-EB** [comment]
Work's like a charm. Thanks @s1monw.
